### PR TITLE
use appointment status field

### PR DESCRIPTION
**Pull request by shwinky**
need to create an index in firebase for the new field so we can query by it...

### CODE REVIEW

File: packages/functions/src/dal/AppointmentDataAccessLayer.ts [hunk 1]
```diff
@@ -121,6 +121,7 @@ export async function getAvailableAppointments() {
     .firestore()
     .collection(Collections.APPOINTMENTS)
     .where("donorId", "==", "")
+    .where("status", "==", AppointmentStatus.AVAILABLE)
```
Comment (yaron1m) [inline packages/functions/src/dal/AppointmentDataAccessLayer.ts:124]: I think this should replace the `donorId` check

File: packages/functions/src/donor/BookAppointmentHandler.ts [hunk 2]
```diff
@@ -2,16 +2,10 @@ import { getDonorOrThrow, setDonor } from "../dal/DonorDataAccessLayer";
...
 } from "../dal/AppointmentDataAccessLayer";
 import * as admin from "firebase-admin";
-import {
-  AppointmentStatus,
-  BookingChange,
-  DbDonor,
-  FunctionsApi,
-  Hospital,
-} from "@zm-blood-components/common";
+import { AppointmentStatus, BookingChange, DbDonor, FunctionsApi, Hospital } from "@zm-blood-components/common";
```
Comment (yaron1m) [inline packages/functions/src/donor/BookAppointmentHandler.ts:8]: This means you ran install in `function` folder and not in the outer one.

run `yarn install` in the root folder, and then `style` again

File: packages/functions/tsconfig.json [hunk 3]
```diff
@@ -1,3 +1,24 @@
 {
-  "extends": "../../tsconfig.json"
+  "extends": "../../tsconfig.json",
```
Comment (yaron1m) [inline packages/functions/tsconfig.json:2]: Please remove these changes

File: packages/functions/src/donor/CancelAppointmentHandler.ts [hunk 4]
```diff
@@ -3,7 +3,7 @@ import {
   removeDonorFromDbAppointment,
   setAppointment,
 } from "../dal/AppointmentDataAccessLayer";
-import { FunctionsApi } from "@zm-blood-components/common";
+import { AppointmentStatus, FunctionsApi } from "@zm-blood-components/common";
```
Comment (yaron1m) [inline packages/functions/src/donor/CancelAppointmentHandler.ts:6]: Is this needed?

**Review by yaron1m — COMMENTED** [review]
The PR looks good but we can't merge it until we run the migration on all existing appointments, right?

**Comment by yaron1m** [comment]
Also, I see tests are failing because there's no index on `status` and `donationStartTime`. You need to add it in STG and Prod

**Comment by shwinky** [comment]
@yaron1m if only you read my first comment :)
need you help creating that index....since it don't know how to :)

**Review by yaron1m — APPROVED** [review]

**Pull request merged**